### PR TITLE
Migrate to sveltekit 2 and update deps

### PR DESCRIPTION
**Pull request by jpnsantoss**

### CODE REVIEW

**Comment by MRita443** [comment]
Currently, this PR is missing two things: @LuisDuarte1 is updating the storybook tests CI to run locally but it is running infinitely. The dependencies upgrade part seems to be done already, but it generating some overriding peer dependency warnings regarding vite that we would like to try to get rid of. @jpnsantoss experimented with `peerDependencies` but to no avail so far.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -67,20 +70,14 @@ jobs:
...
-        id: waitForNetlify
-        with:
-          site_name: 'niaefeup-frontend-storybook'
-          max_timeout: 300 # 5 minutes
-        env:
-          NETLIFY_TOKEN: ${{ secrets.NETLIFY_TOKEN }}
-
       - name: Test
-        run: npm run test-storybook:coverage -- --url ${{ steps.waitForNetlify.outputs.url }} --ci
+        run: npm run test-storybook:ci
```
Comment (BrunoRosendo) [inline .github/workflows/ci.yml:83]: I'm curious as to how this CI works :-)

**Review by BrunoRosendo — COMMENTED** [review]
Nothing to point out, let's just wait for what @MRita443 said

File: .github/workflows/ci.yml (hunk 1 above)
Comment (LuisDuarte1) [reply .github/workflows/ci.yml:83]: https://github.com/storybookjs/test-runner?tab=readme-ov-file#2-running-against-locally-built-storybooks-in-ci

TLDR: it builds the storybook locally, and serves it under a local HTTP server on the CI and then tests it. I don't know why it's hanging indefinitely, yet :sweat_smile:

**Comment by LuisDuarte1** [comment]
The CI is now running the storybook web server locally, however, it's not an optimal CI setup, we could cache the playwright dependencies and cache the storybook-static folder in case there aren't any changes. This is not a priority because we have unlimited minutes so I'll open an issue to optimize the CI.

File: package.json [hunk 2]
```diff
@@ -17,55 +17,59 @@
     "test-vite:watch": "vitest --watch",
     "test-storybook": "test-storybook --browsers firefox",
     "test-storybook:coverage": "npm run test-storybook -- --coverage",
-    "lint": "prettier --plugin-search-dir . --check . && eslint .",
-    "format": "prettier --plugin-search-dir . --write .",
+    "test-storybook:ci": "concurrently -k -s first -n \"SB,TEST\" -c \"magenta,blue\" \"npx http-server storybook-static --port 6006 --silent\" \"npx wait-on tcp:127.0.0.1:6006 && npm run test-storybook:coverage\"",
```
Comment (BrunoRosendo) [inline package.json:20]: I'm a bit lazy so can you please explain this part? `-k -s first -n \"SB,TEST\" -c \"magenta,blue\"` I'm curious 🥺

**Comment by BrunoRosendo** [comment]
> The CI is now running the storybook web server locally, however, it's not an optimal CI setup, we could cache the playwright dependencies and cache the storybook-static folder in case there aren't any changes. This is not a priority because we have unlimited minutes so I'll open an issue to optimize the CI.

It's nice that we have unlimited minutes but it's still a much better dev experience to have the folder cached, it should be a priority imo

File: package.json (hunk 2 above)
Comment (LuisDuarte1) [reply package.json:20]: It's just for styling, it prepends `SB` and `TEST` before each concurrent part and also adds some colors to it :sweat_smile:

**Review by LuisDuarte1 — APPROVED** [review]
Good work :rocket:

**Review by BrunoRosendo — APPROVED** [review]
